pyomo/solvers/plugins/solvers/CPLEX.py

```diff
@@ -185,6 +185,17 @@ def _presolve(self, *args, **kwds):
         self._warm_start_file_name = kwds.pop('warmstart_file', None)
         user_warmstart = False
         if self._warm_start_file_name is not None:
+            if ' ' in self._warm_start_file_name:
+                if self.version()[:2] >= (12,8):
+                    if '"' not in self._warm_start_file_name:
+                        self._warm_start_file_name \
+                            = '"'+self._warm_start_file_name+'"'
+                else:
+                    raise ValueError(
+                        "Space detected in CPLEX warm start file path/name and "
+                        "CPLEX older than version 12.8.  Please either upgrade "
+                        "CPLEX or remove the space from the warm start path.")
+
```

Comment:
It seems that the assumption is, if there is any quote in the path, then we are going to trust the user. Is that correct?

Reply:
Yes. Given @qtothec’s comment, I am going to rewrite the test to disallow anything other than `a-zA-Z0-9.-_`, plus spaces in 12.8+. I think I will only allow quotes as the first and last character.